spv/verifypayment.go

```diff
@@ -4,176 +4,31 @@ import (
 	"context"
 
 	"github.com/libsv/go-bt/v2"
-	"github.com/pkg/errors"
 )
 
-// VerifyPayment verifies whether or not the txs supplied via the supplied spv.Envelope are valid.
-// This method also accepts functional options which will override the options set in the verifier upon creation.
-// Any options passed here persist only for that call and do not override the main options. This gives flexibility
-// in that you can setup a verifier with sensible defaults and override them conditionally by providing
-// options here to enable or disable checks as required.
-//
-//  // with functional options to override verifier defaults.
-//  tx, err := VerifyPayment(ctx, env, spv.VerifyProofs(), spv.VerifyFees(fees))
-//
-//  // without functional options to use verifier defaults.
-//  tx, err := VerifyPayment(ctx, env)
-//
-func (v *verifier) VerifyPayment(ctx context.Context, initialPayment *Envelope, opts ...VerifyOpt) (*bt.Tx, error) {
-	if initialPayment == nil {
-		return nil, ErrNilInitialPayment
+// VerifyPaymentWithAncestry is a method for parsing a binary payment transaction and its corresponding ancestry in binary.
+// It will return the paymentTx struct if all validations pass.
+func (v *verifier) VerifyPaymentWithAncestry(ctx context.Context, pTx []byte, ancestors []byte, opts ...VerifyOpt) (*bt.Tx, error) {
+	o := &verifyOptions{
+		proofs: true,
+		script: true,
+		fees:   false,
 	}
-	vOpt := v.opts.clone()
```

Comment:
Is there a reason default overriding is being removed?

The idea behind the options were, you create your verifier with a set of opts, which is your default config to be applied to all verifications.

However, in this function, we're extending the option to override your default config, so for example, by default you may set script verification to be execute, but if the tx you're verifying is less than 500 satoshis, we may decide it's not worth the computing power to execute the scripts, so we could override it on that single verification

Reply:
Good catch, it doesn't make sense to me either why it was removed. I think probably just fat fingered it.